### PR TITLE
Initial implementation of an EZproxy ticket authentication handler. T…

### PR DESCRIPTION
**Pull request by EreMaijala**
…his allows a user to be authorized into EZproxy via VuFind login.

TODO:
- [x] Wiki page for documentation

### CODE REVIEW

**Comment by EreMaijala** [comment]
It's pretty difficult to test this one without EZproxy, and the latest versions require a key to work. It's possible to request an evaluation key from OCLC, but it may take a good while for them to process the request. If it helps, this is direct copy from our production code and tested to work properly.

File: themes/bootstrap3/templates/externalauth/ezproxylogin.phtml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+<div class="row external-content-access">
+  <div class="col-sm-12">
+    <?=$this->flashmessages()?>
+    <? if ($this->unauthorized): ?>
+      <div class="unauthorized-description">
+        <p><?=$this->transEsc('external_auth_unauthorized_desc'); ?></p>
+      </div>
+      <div>
+        <a href="<?=$this->url('myresearch-logout')?>" class="logout<?=$logoutClass?> btn btn-primary" title="<?=$this->transEsc("Log Out")?>"><strong><?=$this->transEsc("Log Out")?></strong></a>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/externalauth/ezproxylogin.phtml]: What is the purpose of the $logoutClass variable here? Is that something specific to your local instance?

Comment (EreMaijala) [inline themes/bootstrap3/templates/externalauth/ezproxylogin.phtml]: Yes. I tried to get rid of all our local stuff but missed that one.

**Comment by EreMaijala** [comment]
Thanks! Wiki page created and linked to from the configuration page. Feel free to review and tweak as necessary.

**Comment by demiankatz** [comment]
Ere,

Thanks, I have merged this.

One possible uncovered use case occurred to me. Right now, when you set the EZproxy/host setting, links are always prefixed. However, I can imagine a situation where a library might want to use VuFind for EZproxy authentication without this side effect (in our situation, for example, the proxy prefixes are added in our MARC records and should not be double-added by VuFind). Thus, I added a new setting to turn off prefixing, so it is not the host setting alone that controls this behavior. It defaults to "prefixing on when host set" for backward compatibility. You can find the code in commit https://github.com/vufind-org/vufind/commit/4f25acc57718d5353c52e01168c32578f7ed0832. Please let me know if this causes any problems or raises any concerns.

**Comment by EreMaijala** [comment]
Sounds good to me, and since the default is true, it shouldn't cause trouble. We actually have a more fine-grained control over the prefixing using "include" and "exclude" settings that also allow regexp's. Something to be considered for a pull request if you think there could be use for that (e.g. in your case you could have a rule that anything already prefixed is left alone).

**Comment by demiankatz** [comment]
That's not a feature that we actively need right now, but it's certainly useful in theory. If having it in core would help reduce your maintenance burden, then by all means open a PR. In looking at the proxy view helper code, it also occurs to me that the current approach is quite inflexible, and it might be nice to support other types of proxies more easily (e.g. make URL-encoding behavior and the qurl prefixing configurable too).

**Pull request merged**